eth/gasestimator/gasestimator_rip7560.go

```diff
@@ -0,0 +1,345 @@
+package gasestimator
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+func executeRip7560Validation(ctx context.Context, tx *types.Transaction, opts *Options, gasLimit uint64) (*core.ValidationPhaseResult, *state.StateDB, error) {
+	st := tx.Rip7560TransactionData()
+	// Configure the call for this specific execution (and revert the change after)
+	defer func(gas uint64) { st.ValidationGasLimit = gas }(st.ValidationGasLimit)
+	st.ValidationGasLimit = gasLimit
+
+	// Execute the call and separate execution faults caused by a lack of gas or
+	// other non-fixable conditions
+	var (
+		blockContext = core.NewEVMBlockContext(opts.Header, opts.Chain, nil, opts.Config, opts.State)
+		txContext    = vm.TxContext{
+			Origin:   *tx.Rip7560TransactionData().Sender,
+			GasPrice: tx.GasFeeCap(),
+		}
+
+		dirtyState = opts.State.Copy()
+		evm        = vm.NewEVM(blockContext, txContext, dirtyState, opts.Config, vm.Config{NoBaseFee: true})
+	)
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	go func() {
+		<-ctx.Done()
+		evm.Cancel()
+	}()
+
+	// Gas Pool is set to half of the maximum possible gas to prevent overflow
+	vpr, err := core.ApplyRip7560ValidationPhases(opts.Config, opts.Chain, &opts.Header.Coinbase, new(core.GasPool).AddGas(math.MaxUint64/2), dirtyState, opts.Header, tx, evm.Config, true)
+	if err != nil {
+		if errors.Is(err, vm.ErrOutOfGas) {
+			return nil, nil, nil // Special case, raise gas limit
+		}
+		return nil, nil, err // Bail out
+	}
+	return vpr, dirtyState, nil
+}
+
+func EstimateRip7560Validation(ctx context.Context, tx *types.Transaction, opts *Options, gasCap uint64) (uint64, error) {
+	// Binary search the gas limit, as it may need to be higher than the amount used
+	st := tx.Rip7560TransactionData()
+	totalGasLimit, err := st.TotalGasLimit()
+	if err != nil {
+		return 0, err
+	}
+	gasLimit := totalGasLimit - st.PostOpGas
```

Comment:
Maybe it won't affect the result, but shouldn't it be like this?
```suggestion
	gasLimit := totalGasLimit - st.Gas - st.PostOpGas
```